cosign/README.md

```diff
@@ -0,0 +1,125 @@
+    - Public Key Certificate: An electronic document issued by a CA that binds a public key with an individual’s identity, thus providing proof of the public key’s legitimacy.
+    - Public-key Infrastructure (PKI): A comprehensive framework that includes roles, policies, hardware, software, and procedures for creating, managing, distributing, using, storing, and revoking digital certificates as well as managing public-key encryption.
+
+4. Transparency and Auditability
+    - Identity Log: An online provided record, that maps verified identities to their associated signing keys. Such logs help establish trust by providing a history of key associations.
+    - Artifact Log: An online/offline provided record that contains metadata about artifacts, such as when and by whom they were signed. This ledger aids in auditing and tracing the provenance of digital artifacts.
+
+## Scenarios
+
+`ratify-verifier-go` supports multiple verification scenarios based on different signing methods, artifact types, and trust sources[3]. These scenarios can be categorized into the following main types.
+
+| **Verification Scenario**         | **Purpose**                                                            | **Use Case**                                                                 |
+|-----------------------------------|------------------------------------------------------------------------|-----------------------------------------------------------------------------|
+| **Keyless Verification**          | Verifyies signatures from Rekor, trust material (signed certificate timestamp) from the CT log, and certificates that chain up to Fulcio. | CI/CD pipelines or automated workflows where no private key management is needed. |
+| **Key-Based Verification**        | Verifies signature using a known **public key**.                        | Environments where signatures are verified with a known public key. |
+| **Timestamp Verification**        | Verifies the **timestamp** of the signature to prevent time-based attacks. Not standalone verification. | Long-term signature validity checks, ensuring signatures are valid at a specific point in time. |
+| **Rekor Transparency Log (TLog)** | Verifies inclusion of the signature in the **Rekor Transparency Log** for audibility. Not standalone verification. | Auditing and compliance to ensure signatures are publicly recorded in an immutable log. |
+
+### Cosign Verifier Input
+
+1. The Artifact to Verify
+    - The artifact in OCI-compliant registries.
+
+2. Verification Materials
+    - The signature(s) of the artifact[4]: The cryptographic signature(s) created when the artifact was signed.
+    - Verification options: i.e. whether to expect signed certificate timestamps (SCTs), TLog entries, or signed timestamps and expected identity and digest to verify.
+
+3. Verification Key or Certificate
+    - Keyless Verification: Retrieves an X.509[5] certificate on the signature and verify against Fulcio root trust.
```

Comment:
Where do we find the "Fulcio root trust"?

Reply:
[https://fulcio.sigstore.dev](https://github.com/sigstore/sigstore-go/blob/main/examples/trusted-root-public-good.json)